orchagent/intfsorch.cpp

```diff
@@ -99,8 +102,20 @@ IntfsOrch::IntfsOrch(DBConnector *db, string tableName, VRFOrch *vrf_orch, DBCon
         SWSS_LOG_WARN("RIF flex counter group plugins was not set successfully: %s", e.what());
     }
 
-    if(gMySwitchType == "voq")
+    if (gMySwitchType == "voq")
     {
+        //STATE DB connection for setting state of the VoQ inband VLAN interface
+        unique_ptr<DBConnector> stateDb;
+        stateDb = make_unique<DBConnector>("STATE_DB", 0);
+        m_stateVlanTable = unique_ptr<Table>(new Table(stateDb.get(), STATE_VLAN_TABLE_NAME));
+
+        //Timer to bring up inband Vlan netdev
+        auto intervT = timespec { .tv_sec = INBAND_INTF_TIMER_SEC , .tv_nsec = 0 };
+        m_inbandVlanTimer = new SelectableTimer(intervT);
```

Comment:
why do we need timer ? Can you point to design doc that cover this ?

Reply:
Since SAI create call is not synchronized, we cannot bring up inband vlan in kernel immediately after creating inband vlan with SAI call.  We use this timer to periodically check if it's ready to bring up  inband vlan in kernel or not. The timer is stopped once inband vlan is brought up in kernel successfully. I think this is implementation details, and thus is not covered in HLD.

Reply:
@ysmanman how is this different w.r.t Regular VLAN IP Interface ? We don;t need timer there.

Reply:
@abdosi There are few differences between regular Vlan and Inban Vlan:
0) The members of Inband Vlan are local and remote CPU ports, which don't have kernel devices created today.
1) Inband Vlan is a static configuration and imembership is unlikely user configurable.
These are the main reasons that Inband Vlan was implemented different from regular Vlans and therefore we added the timer.

Reply:
sai call is synchronized now starting from 202012 release